packages/odf/components/s3-browser/upload-objects/uploads.ts

```diff
@@ -0,0 +1,82 @@
+import * as React from 'react';
+import { Upload } from '@aws-sdk/lib-storage';
+import { S3Commands } from '@odf/shared/s3';
+import * as _ from 'lodash-es';
+import { UploadProgress, UploadStatus } from './types';
+
+export function* uploadFile(
+  files: File[],
+  client: S3Commands,
+  bucketName: string,
+  setUploadProgress: React.Dispatch<React.SetStateAction<UploadProgress>>
+): Generator<unknown, unknown, ReturnType<typeof Upload.prototype.done>[]> {
+  const performUploadPromise = (
+    file: File
+  ): ReturnType<typeof Upload.prototype.done> => {
+    const uploader = new Upload({
+      client: client,
+      params: {
+        Bucket: bucketName,
+        Key: file.name,
```

Comment:
u need to get current `foldersPath` (which signifies ur current folder inside the bucket) and append that to the file name... this will "logically" or "virtually" make sure that these files are uploaded inside a selected folder...
```suggestion
        Key: getPrefix(file.name, foldersPath),
```

Reply:
we need `file.webkitRelativePath` to make sure a folder is uploaded when selected by user...